include/libtorrent/aux_/win_util.hpp

```diff
@@ -38,16 +38,40 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent { namespace aux {
 
+#ifdef TORRENT_WINRT
+	using LoadLibraryAPrototype = HMODULE WINAPI(_In_ LPCSTR lpLibFileName);
+#endif
+
 	template <typename Library>
 	HMODULE get_library_handle()
 	{
 		static bool handle_checked = false;
-		static HMODULE handle = nullptr;
+		static HMODULE handle;
 
 		if (!handle_checked)
 		{
-			handle = LoadLibraryA(Library::library_name);
 			handle_checked = true;
+
+#ifdef TORRENT_WINRT
+			MEMORY_BASIC_INFORMATION Information;
+
+			if (::VirtualQuery(&VirtualQuery, &Information, sizeof(Information)) == 0)
+			{
+				return nullptr;
+			}
+
+			const auto SyscallBegin = static_cast<HMODULE>(Information.AllocationBase);
```

Comment:
so this is, essentially, the handle to `kernel32` then?

Reply:
Yeah, the pointer ends up in KernelBase.dll